core/tauri/src/state.rs

```diff
@@ -19,6 +19,13 @@ impl<'r, T: Send + Sync + 'static> State<'r, T> {
   pub fn inner(&self) -> &'r T {
     self.0
   }
+
+  /// Create a State for testing commands in unittests
+  #[cfg(test)]
```

Comment:
This makes the `new_testonly` function only available to the Tauri crate itself. Here's how rocket handles it: https://api.rocket.rs/v0.4/rocket/struct.State.html#testing-with-state
We do have some test helpers to create a dummy App instance for testing, we should expose that instead.

Reply:
State::from is not as good an approach because it exposes a new API that means it is a breaking change if State ever ends up holding other things internally. A dummy App might be an OK solution too, though.

Reply:
Let's track that over #4752